chia/consensus/blockchain.py

```diff
@@ -932,7 +940,7 @@ async def get_block_records_at(self, heights: list[uint32], batch_size: int = 90
         return records
 
     def try_block_record(self, header_hash: bytes32) -> Optional[BlockRecord]:
-        if self.contains_block(header_hash):
+        if header_hash in self.__block_records:
             return self.block_record(header_hash)
```

Comment:
it seems a bit backwards to make `try_block_record()` be implemented in terms of `block_record()`. It (probably) would be simpler to do it the other way around.

Reply:
i dont have a strong opinion on that, if we do it the other way around we need to throw an explicit exception for when we dont have the block